src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -385,6 +397,26 @@ def disks_for(devicegraph, device_name = nil)
       def candidate_disk_names
         settings.candidate_devices
       end
+
+      # Whether {#resize_and_delete} should be executed several times,
+      # see {#calculate_new_graph} for details.
+      #
+      # @param parts_by_disk [Hash{String => Array<Planned::Partition>}] see
+      #   {#planned_partitions_by_disk}
+      # @return [Boolean]
+      def several_passes?(parts_by_disk)
+        # In this case the result is not much relevant since #resize_and_delete
+        # wouldn't be executed for particular disks anyway
+        return false if parts_by_disk.empty?
+
+        return true if parts_by_disk.size > 1
+
+        # In theory, the specific disk mentioned in the planned partitions
+        # (note that, at this point, we are sure there is only one) should be
+        # included in the set of candidate disks. Return false if that's not the
+        # case or if there is more than one candidate disk.
+        parts_by_disk.keys != candidate_disk_names
```

Comment:
Why cannot there be several candidates? I guess we can have several candidate devices but current planning is using only one. I guess I am missing something.

Reply:
Yes, I am fine with your solution. But, thinking again on it, maybe the following is enough:

```
if candidate_disks.size > 1
  planned_partitions_by_disk(partitions).each do |disk, parts|
    resize_and_delete(parts, keep, lvm_helper, disk_name: disk)
  end
end

resize_and_delete(partitions, keep, lvm_h
```

Yes, this solution is only valid when the keys of `parts_by_disks` are always included in the `candidate_disk_names` (and this should be always the case), but due to  the code was not ensuring it, I see your proposal a better solution to ensure the same behavior as before.